src/main/java/com/amazonaws/kinesisvideo/java/mediasource/file/ImageFrameSource.java

```diff
@@ -90,6 +91,15 @@ private void generateFrameAndNotifyListener() throws KinesisVideoException {
                 if (isMetadataReady()) {
                     mkvDataAvailableCallback.onFragmentMetadataAvailable(metadataName + metadataCount,
                             Integer.toString(metadataCount++), false);
+                    
+                    if (isKeyFrame()) {
+                        // Put event metadata on keyframes.
+                        StreamEventMetadata eventMetadata;
+                        String[] eventMetadataNames = new String[]{"eventMetadata-name-1"};
```

Comment:
I would not recommend hard coding these values. Was this done just to test the feature?

Reply:
Yes, haven't put much thought into how we will leave this in the sample, I'll add some macros in the main demo file to use here as a start.

Reply:
Ended up inializing these at the top to be uniform with how the fragmentMetadata values are.